lib/remap-cli/src/main.rs

```diff
@@ -0,0 +1,120 @@
+use remap::{state, Object, Program, Runtime, Value};
+use std::collections::BTreeMap;
+use std::fs::File;
+use std::io::{self, Read};
+use std::iter::IntoIterator;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "TRL", about = "Timber Remap Language CLI")]
+struct TRL {
+    /// Program to execute
+    ///
+    /// For example, ".foo = true" will set the object's `foo` field to `true`.
+    #[structopt(name = "PROGRAM")]
+    program: Option<String>,
+
+    /// File containing the object to manipulate, leave empty to use stdin
+    #[structopt(short, long = "input", parse(from_os_str))]
+    input_file: Option<PathBuf>,
+
+    /// File containing the program to execute, can be used instead of
+    /// "PROGRAM"
+    #[structopt(short, long = "program", conflicts_with("program"), parse(from_os_str))]
+    program_file: Option<PathBuf>,
+
+    /// Print the (modified) object, instead of the result of the final
+    /// expression.
+    ///
+    /// The same result can be achieved by using `.` as the final expression.
+    #[structopt(short = "o", long)]
+    print_object: bool,
+}
+
+#[derive(thiserror::Error, Debug)]
+enum Error {
+    #[error("io error")]
+    Io(#[from] io::Error),
+
+    #[error("remap error: {0}")]
+    Remap(#[from] remap::RemapError),
```

Comment:
TRL error improvements are on my list, at which time this will simply be `remap::Error` and doesn't require the `{0}` on the line above — don't ask why this is currently needed, it's complicated (read: I should've RTFM) 😞.